shared.go

```diff
@@ -118,6 +119,43 @@ type Value struct {
 	Name string `json:"name"`
 }
 
+// Field is generic Name/Value struct applied to a few places.
+type Field struct {
+	Advanced                    bool            `json:"advanced,omitempty"`
+	Order                       int64           `json:"order,omitempty"`
+	HelpLink                    string          `json:"helpLink,omitempty"`
+	HelpText                    string          `json:"helpText,omitempty"`
+	Hidden                      string          `json:"hidden,omitempty"`
+	Label                       string          `json:"label,omitempty"`
+	Name                        string          `json:"name"`
+	SelectOptionsProviderAction string          `json:"selectOptionsProviderAction,omitempty"`
+	Type                        string          `json:"type,omitempty"`
+	Value                       interface{}     `json:"value,omitempty"`
+	SelectOptions               []*SelectOption `json:"selectOptions,omitempty"`
+}
+
+func (f *Field) GetFieldValueString() (string, error) {
+	if v, a := f.Value.(string); !a {
+		return v, nil
+	}
+	return "", fmt.Errorf("Field %s has invalid value", f.Name)
```

Comment:
There's a couple problem with this approach. Since this is a dynamic error, the caller can never use `errors.Is` to test it. The caller could just do the type assertion themselves, and handle it however they want anyway, which makes the usefulness of this method questionable. I know I recommended this, but as I see it now it seems less useful.

Reply:
Yes, I struggled a bit with this one. I think it is needed if we define the fields as user struct output, whereas it's not if we use a generic interface... For sure if we keep the interface, and avoid all this, the SDK implementation will be easier and cleaner. Also, a meaningful error will still be sent back in API response. However, we'll burden the user with the type assertion, and they must know which fields are required and which are optional (which is fine for me). If you prefer to proceed this way, It's fine for me.

Reply:
> they must know which fields are required and which are optional 

Indeed, and this is sort of true regardless of how we implement our interface.

Reply:
Ok, let's keep that easy and clean. I'll change this accordingly